### PR TITLE
Added struct and record class/record struct serialization/deserialization

### PR DESCRIPTION
**Pull request by jvbsl**
* Added test file for struct serialization/deserialization
* Added test file for record serialization/deserialization
* Enabled attribute usage for struct types in NoosonAttribute
* Implemented basic struct, record class and record struct serialization/deserialization
* Updated VaVare for record and struct builders

### CODE REVIEW

File: DEMO/StructTestFile.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+using NonSucking.Framework.Serialization;
+
+namespace DEMO;
+
+
```
Comment (Gallimathias) [inline DEMO/StructTestFile.cs]:
```suggestion
```

File: NonSucking.Framework.Serialization/NoosonGenerator.cs [hunk 2]
```diff
@@ -179,36 +180,53 @@ var attribute
...
             }
-            MemberInfo[] properties
-                = Helper
-                    .GetMembersWithBase(typeSymbol)
-                    .Where(x=>!x.Symbol.ContainingType.IsAbstract)
-                    .ToArray();
+
+            var propEnumerable = Helper
+                .GetMembersWithBase(typeSymbol)
+                .Where(x => !x.Symbol.ContainingType.IsAbstract);
```
Comment (Gallimathias) [inline NonSucking.Framework.Serialization/NoosonGenerator.cs]:
```suggestion
            var propEnumerable 
                = Helper
                .GetMembersWithBase(typeSymbol)
                .Where(x => !x.Symbol.ContainingType.IsAbstract);
```

File: NonSucking.Framework.Serialization/NoosonGenerator.cs [hunk 3]
```diff
@@ -179,36 +180,53 @@ var attribute
...
-                    .Build();
-
-                string hintName = $"{classToAugment.TypeSymbol.ToDisplayString()}.Nooson.cs";
+                CompilationUnitSyntax sourceCode;
+                if (typeToAugment.TypeSymbol.IsRecord)
+                    sourceCode = new RecordBuilder(typeToAugment.TypeSymbol.Name, typeToAugment.TypeSymbol.ContainingNamespace.ToDisplayString(), typeToAugment.TypeSymbol.IsValueType)
+                        .WithUsings()
+                        .WithModifiers(Modifiers.Public, Modifiers.Partial)
+                        .WithMethods(methods)
+                        .Build();
```
Comment (Gallimathias) [inline NonSucking.Framework.Serialization/NoosonGenerator.cs]:
```suggestion
                    sourceCode 
                        = new RecordBuilder(typeToAugment.TypeSymbol.Name, typeToAugment.TypeSymbol.ContainingNamespace.ToDisplayString(), typeToAugment.TypeSymbol.IsValueType)
                        .WithUsings()
                        .WithModifiers(Modifiers.Public, Modifiers.Partial)
                        .WithMethods(methods)
                        .Build();
```

File: NonSucking.Framework.Serialization/NoosonGenerator.cs [hunk 4]
```diff
@@ -179,36 +180,53 @@ var attribute
...
+                else
+                    sourceCode = new ClassBuilder(typeToAugment.TypeSymbol.Name, typeToAugment.TypeSymbol.ContainingNamespace.ToDisplayString())
+                        .WithUsings()
+                        .WithModifiers(Modifiers.Public, Modifiers.Partial)
+                        .WithMethods(methods)
+                        .Build();
+                string hintName = $"{typeToAugment.TypeSymbol.ToDisplayString()}.Nooson.cs";
```
Comment (Gallimathias) [inline NonSucking.Framework.Serialization/NoosonGenerator.cs]:
```suggestion
                else if (typeToAugment.TypeSymbol.IsValueType)
                    sourceCode 
                        = new StructBuilder(typeToAugment.TypeSymbol.Name, typeToAugment.TypeSymbol.ContainingNamespace.ToDisplayString())
                        .WithUsings()
                        .WithModifiers(Modifiers.Public, Modifiers.Partial)
                        .WithMethods(methods)
                        .Build();
                else
                    sourceCode 
                        = new ClassBuilder(typeToAugment.TypeSymbol.Name, typeToAugment.TypeSymbol.ContainingNamespace.ToDisplayString())
                        .WithUsings()
                        .WithModifiers(Modifiers.Public, Modifiers.Partial)
                        .WithMethods(methods)
                        .Build();
                        
                string hintName = $"{typeToAugment.TypeSymbol.ToDisplayString()}.Nooson.cs";
```

File: NonSucking.Framework.Serialization/Templates/Attribute/NoosonAttribute.cs [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 namespace NonSucking.Framework.Serialization
 {
-    [AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct, Inherited = false, AllowMultiple = false)]
```
Comment (Gallimathias) [inline NonSucking.Framework.Serialization/Templates/Attribute/NoosonAttribute.cs:6]: Gibts kein Targets für Records?

**Review by Gallimathias — CHANGES_REQUESTED** [review]

File: NonSucking.Framework.Serialization/Templates/Attribute/NoosonAttribute.cs (hunk 5 above)
Comment (jvbsl) [reply NonSucking.Framework.Serialization/Templates/Attribute/NoosonAttribute.cs:6]: Nein :D

**Review by Gallimathias — APPROVED** [review]

**Pull request merged**